### ISSUE TITLE
save authorization and assigned event across restarts

### ISSUE THREAD

**Issue by HarHarLinks** [MoralCode/maubot-events#5]
when i update the plugin etc, rooms should stay connected

**Comment by MoralCode** [comment]
yeah this is something I plan to add - probably next week or so. Its also been annoying for me when the bot restarts during testing

**Comment by MoralCode** [comment]
auth is now being persisted. not the event -> room mappings tho

**Comment by HarHarLinks** [comment]
cool! reauthenticating is the most tedious part. now the mappings should not be much harder :)

personally i like to store data that belongs to the room directly in the room state. obviously that isn't an option for secrets such as the auth token.

have you looked into the DB api? https://docs.mau.fi/maubot/dev/database/index.html that might make it fairly easy to scale the required storage flexibly.

**Comment by MoralCode** [comment]
the auth is (essentially) global so that shouldnt be needed per-room. although it may need some changes if this is going to be used in an environment with multiple orgs hosting events with the same bot isntance

Room state defintely sounds like a useful place to store the room->event mapping - I didnt know that was a thing you could do (am relatively new to the inner workings of matrix)
